webapp/tests/test_functions.py

```diff
@@ -866,6 +866,38 @@ def test_consolidateBy(self):
             results = functions.consolidateBy({}, seriesList, func)
             self._verify_series_consolidationFunc(results, func)
 
+    def test_smooth(self):
+        seriesList = [
+            TimeSeries('collectd.test-db1.load.value',0,1,1,range(20)),
+        ]
+
+        def mock_evaluateTokens(reqCtx, tokens, replacements=None):
+            seriesList = [
+                TimeSeries('collectd.test-db1.load.value',0,1,1,range(20)),
+            ]
+            for series in seriesList:
+                series.pathExpression = series.name
+            return seriesList
+
+        expectedResults = [
+            TimeSeries('smooth(collectd.test-db1.load.value, 5)',0,1,1,[None,None,None,None,2,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20])
+        ]
+
+        with patch('graphite.render.functions.evaluateTokens', mock_evaluateTokens):
+            result = functions.smooth(
+                {
+                    'template': {},
+                    'args': ({},{}),
+                    'startTime': datetime(1970, 1, 1, 0, 0, 0, 0, pytz.timezone(settings.TIME_ZONE)),
+                    'endTime': datetime(1970, 1, 1, 0, 9, 0, 0, pytz.timezone(settings.TIME_ZONE)),
+                    'localOnly': False,
+                    'width': 400,
+                    'data': []
+                },
+                seriesList, 5
+            )
```

Comment:
@obfuscurity If the test was changed like this, it returns success, but I'm not 100% sure that it does as expected.  The fact that certain inputs can result in no data being returned is concerning.  Also, these inputs only hit the positive case on the `if (len(series) < expectedSamples * 0.95):` on line 1190 of functions.py
```
diff --git a/webapp/tests/test_functions.py b/webapp/tests/test_functions.py
index 5494b7d..84814a7 100644
--- a/webapp/tests/test_functions.py
+++ b/webapp/tests/test_functions.py
@@ -868,19 +868,19 @@ class FunctionsTest(TestCase):
 
     def test_smooth(self):
         seriesList = [
-            TimeSeries('collectd.test-db1.load.value',0,1,1,[range(20)]),
+            TimeSeries('collectd.test-db1.load.value',0,20,60,range(0, 1200, 60)),
         ]
 
         def mock_evaluateTokens(reqCtx, tokens, replacements=None):
             seriesList = [
-                TimeSeries('collectd.test-db1.load.value',0,1,1,[range(20)]),
+                TimeSeries('collectd.test-db1.load.value',0,100,12,range(0, 1200, 12)),
             ]
             for series in seriesList:
                 series.pathExpression = series.name
             return seriesList
 
         expectedResults = [
-            TimeSeries('smooth(collectd.test-db1.load.value, 5)',0,1,1,[None,None,None,None,2,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20])
+            TimeSeries('smooth(collectd.test-db1.load.value, 5)',60,100,12,range(0,1177,12)),
         ]
 
         with patch('graphite.render.functions.evaluateTokens', mock_evaluateTokens):
@@ -889,13 +889,16 @@ class FunctionsTest(TestCase):
                     'template': {},
                     'args': ({},{}),
                     'startTime': datetime(1970, 1, 1, 0, 0, 0, 0, pytz.timezone(settings.TIME_ZONE)),
-                    'endTime': datetime(1970, 1, 1, 0, 9, 0, 0, pytz.timezone(settings.TIME_ZONE)),
+                    'endTime': datetime(1970, 1, 1, 0, 20, 0, 0, pytz.timezone(settings.TIME_ZONE)),
                     'localOnly': False,
-                    'width': 400,
+                    'width': 100,
                     'data': []
                 },
                 seriesList, 5
```